Resources/public/js/views/services/plugins/ez-contentsetmainlocationplugin.js

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-contentsetmainlocationplugin', function (Y) {
+    "use strict";
+    /**
+     * Provides the plugin for set the main location
+     *
+     * @module ez-contentsetmainlocationplugin
+     */
+    Y.namespace('eZ.Plugin');
+
+    /**
+     * Set main location plugin. It sets an event handler to set the given location as
+     * main location of given content.
+     *
+     * In order to use it you need to fire `setMainLocation` event with parameter
+     * `locationId` containing the location id which will be set as main location of content.
+     *
+     * @namespace eZ.Plugin
+     * @class ContentSetMainLocation
+     * @constructor
+     * @extends eZ.Plugin.ViewServiceBase
+     */
+    Y.eZ.Plugin.ContentSetMainLocation = Y.Base.create('contentsetmainlocationplugin', Y.eZ.Plugin.ViewServiceBase, [], {
+
+        initializer: function () {
+            this.onHostEvent('*:setMainLocation', this._setMainLocationConfirm);
+        },
+
+        /**
+         * setMainLocation event handler, opens confirm box to confirm that given location
+         * to be set as main location
+         *
+         * @method _setMainLocationConfirm
+         * @private
+         * @param {EventFacade} e setMainLocation event facade
+         */
+        _setMainLocationConfirm: function (e) {
+            var service = this.get('host');
+
+            service.fire('confirmBoxOpen', {
+                config: {
+                    title: "Are you sure you want to set this location as main location of content?",
+                    confirmHandler: Y.bind(function () {
+                        this._setMainLocation(e.locationId, e.afterSetMainLocationCallback);
+                    }, this)
+                }
+            });
+        },
+
+        /**
+         * Sets the given location as the main location of content. After that calls the callback function.
+         *
+         * @method _setMainLocation
+         * @protected
+         * @param {String} locationId
+         * @param {Function} callback
+         */
+        _setMainLocation: function (locationId, callback) {
+            var service = this.get('host'),
+                capi = service.get('capi'),
+                contentService = capi.getContentService(),
+                content = service.get('content'),
+                notificationIdentifier = 'set-main-location-' + content.get('id') + '-' + locationId,
+                updateStruct = contentService.newContentMetadataUpdateStruct(content.get('mainLanguageCode')),
+                that = this;
+
+            updateStruct.body.ContentUpdate.MainLocation = {_href: locationId};
```

Comment:
here as well, the JS REST client should be improved to provide a `setMainLocation` (it already has a `setSection` https://github.com/ezsystems/ez-js-rest-client/blob/master/src/structures/ContentMetadataUpdateStruct.js#L27)